src/coreclr/src/jit/simd.cpp

```diff
@@ -97,7 +98,9 @@ int Compiler::getSIMDTypeAlignment(var_types simdType)
         return 32;
     }
 #elif defined(TARGET_ARM64)
-    return 16;
+    // preferred alignment for 64-bit vectors is 8-bytes.
+    // For everything else, 16-bytes.
+    return (size == 8) ? 8 : 16;
```

Comment:
For reference, the actual alignment of the ABI types is here: <https://github.com/dotnet/runtime/blob/master/src/coreclr/src/vm/methodtablebuilder.cpp#L9541-L9580>
We can't necessarily get this info from the VM because `Vector2`/`Vector3`/`Vector4`, and `Vector<T>` aren't classified as the ABI types, but we still align them as such if they are on the stack.